test/e2e/e2e_test.go

```diff
@@ -115,13 +129,68 @@ do
 	done                                                         
 	echo "Successfully pinged pod: $pod"
 done
-sleep 30`, podHostnames[0], podHostnames[1], podHostnames[2], podHostnames[3])
+sleep 30`, hostnames[0], hostnames[1], hostnames[2], hostnames[3])
+}
 
+// 1 replicated job with 4 replicas, DNS hostnames enabled
+func pingTestJobSet(ns *corev1.Namespace) *testing.JobSetWrapper {
+	jsName := "js"
+	rjobName := "rjob"
+	replicas := 4
+	var podHostnames []string
+	for jobIdx := 0; jobIdx < replicas; jobIdx++ {
+		// Pod hostname format:
+		// <jobSet.name>-<spec.replicatedJob.name>-<job-index>-<pod-index>.<jobSet.name>
+		podHostnames = append(podHostnames, fmt.Sprintf("%s-%s-%d-0.%s", jsName, rjobName, jobIdx, jsName))
+	}
+	cmd := getPingCommand(podHostnames)
+	return testing.MakeJobSet(jsName, ns.Name).
+		EnableDNSHostnames(true).
+		NetworkSubdomain(jsName).
+		// We have to explicitly set it since the webhook does not work
+		Subdomain(jsName).
+		ReplicatedJob(testing.MakeReplicatedJob(rjobName).
+			Job(testing.MakeJobTemplate("job", ns.Name).
+				PodSpec(corev1.PodSpec{
+					RestartPolicy: "Never",
+					Subdomain:     jsName,
+					Containers: []corev1.Container{
+						{
+							Name:    "ping-test-container",
+							Image:   "bash:latest",
+							Command: []string{"bash", "-c"},
+							Args:    []string{cmd},
+						},
+					},
+				}).Obj()).
+			Replicas(replicas).
+			Obj())
+}
+
+// 1 replicated job with 4 replicas, DNS hostnames + subdomain enabled
+func pingTestJobSetSubdomain(ns *corev1.Namespace) *testing.JobSetWrapper {
+	jsName := "js"
+	rjobName := "rjob"
+	replicas := 4
+	subdomain := "network-subdomain"
+	var podHostnames []string
+	for jobIdx := 0; jobIdx < replicas; jobIdx++ {
+		// Pod hostname format:
+		// e.g.,js-rjob-0-0.network-subdomain.e2e-7vd7z.svc.cluster.local       js-rjob-0-0
+		// <jobSet.name>-<spec.replicatedJob.name>-<job-index>-<pod-index>.<subdomain>
+		podHostnames = append(podHostnames, fmt.Sprintf("%s-%s-%d-0.%s", jsName, rjobName, jobIdx, subdomain))
+	}
+	cmd := getPingCommand(podHostnames)
 	return testing.MakeJobSet(jsName, ns.Name).
+		EnableDNSHostnames(true).
+		NetworkSubdomain(subdomain).
+		// We have to explicitly set it since the webhook does not work
+		Subdomain(subdomain).
 		ReplicatedJob(testing.MakeReplicatedJob(rjobName).
 			Job(testing.MakeJobTemplate("job", ns.Name).
 				PodSpec(corev1.PodSpec{
 					RestartPolicy: "Never",
+					Subdomain:     subdomain,
```

Comment:
we shouldn't be setting this explicitly, jobset does that.

Reply:
Will get in a follow-up PR soon!